bundles/http/Citrix/0b4d1a6b-30f3-4134-a354-42fa008b757e/f3b61af4-efd0-43c2-9a08-a467fc8242c1.js

```diff
@@ -0,0 +1,1029 @@
+        }).map(value =>{
+            let com = []
+            if(value.$.StatusName == "SUBMITTED"){
+                com = value.Comments!=undefined?value.Comments[0].Comment[0]:[]
+            }
+            else if(value.$.StatusName == "APPROVED" || value.$.StatusName == "REFUSED"){
+                com = value.Comments!=undefined?value.Comments[0].Comment[value.Comments[0].Comment.length - 1]:[]    
+            }
+            return {
+                "endDate":moment(value.$.EndDate,`${integrationParameters.dateFormat}`).format('yyyy-MM-DD'),
+                "endTime":value.$.EndTime,
+                "startDate":moment(value.$.StartDate,`${integrationParameters.dateFormat}`).format('yyyy-MM-DD'),
+                "startTime":value.$.StartTime,
+                "personNumber":pno.personNumber,
+                "StatusName":value.$.StatusName,
+                "Length":value.$.Length,
+                "requestId":value.$.Id,
+                "CommentText":com.$ !=undefined ?com.$.CommentText:null,
+                "CommentDataSourceClientName":com.$ !=undefined ?com.Notes[0].Note[0].DataSource[0].DataSource[0].$.ClientName:null,
+                "CommentDataSourceFunctionalAreaCode":com.$ !=undefined ?Number(com.Notes[0].Note[0].DataSource[0].DataSource[0].$.FunctionalAreaCode):null,
+                "CommentDataSourceFunctionalAreaName":com.$ !=undefined ?com.Notes[0].Note[0].DataSource[0].DataSource[0].$.FunctionalAreaName:null,
+                "CommentDataSourceServerName":com.$ !=undefined ?com.Notes[0].Note[0].DataSource[0].DataSource[0].$.ServerName:null,
+                "CommentDataSourceUserName":com.$ !=undefined ?com.Notes[0].Note[0].DataSource[0].DataSource[0].$.UserName:null,
+                "CommentNotesText":com.$ !=undefined ?com.Notes[0].Note[0].$.Text:null,
+                "CommentNotesTimestamp":com.$ !=undefined ?com.Notes[0].Note[0].$.Timestamp:null,
+                "CreationDateTime":moment(value.$.CreationDateTime, `${integrationParameters.dateFormat} hh:mm A`).toDate(),
+            }
+        })
+        dataStore.save('work_time',syncWorktimeData)
+    }
+}
+
+function getValidDateFormat(dateFormat){
+    let validFormats = new Set(['YYYY/MM/DD', 'MM/DD/YYYY', 'YYYY-MM-DD', 'MM-DD-YYYY','M-DD-YYYY', 'M/DD/YYYY', 'DD-MM-YYYY'])
+    if(!validFormats.has(dateFormat.toUpperCase())){
+        throw new Error(`Date Format [${dateFormat}] is not supported , choose any of the supported formats [${[...validFormats].toString()}] `)
+    }
+    return dateFormat.toUpperCase()
+}
+
+async function incrementalSync({ dataStore, client,latestSynchronizationTime, integrationParameters}){
+    integrationParameters.dateFormat = getValidDateFormat(integrationParameters.dateFormat)
+    const session = await logIn(client, integrationParameters.username , integrationParameters.password);
+    const people = await syncPeople(dataStore, client, session,integrationParameters);
```

Comment:
Yes, I have the same question, as I mention it in my comment in syncPeople function.

Reply:
We have restricted the QueryDateSpan to the current date for incremental Sync functions.
Updated code using last Sync Time:
`const dateSpan = moment(latestSynchronizationTime).format(integrationParameters.dateFormat)`

Reply:
While testing this approach of restricting QueryDateSpan to current date. We find that the QueryDateSpan represents the StartDate of the request instead of creationDateTime. So for incremental sync we need to maintain the same QueryDateSpan as of full sync.

Reply:
Please then just confirm that the incremental won't take too much time in case of thousands of employees.

Reply:
As per the current scenario, for 89 employees it is taking 9 sec on an average for incremental sync.
we have calculated for 10000 employee it will take 15-17 min(approx) for incremental sync.